Composer/packages/lib/shared/src/types/settings.ts

```diff
@@ -33,9 +33,13 @@ export interface DialogSetting {
   defaultLanguage: string;
   languages: string[];
   skill?: {
-    name: string;
-    manifestUrl: string;
-  }[];
+    [skillName: string]: {
+      name: string;
+      manifestUrl: string;
+      msAppId?: string;
```

Comment:
I think `msAppId` and `endpointUrl` should both be optional or both required since they go hand-in-hand.

Reply:
This is basically in the settings. We cannot enforce those two props . But it makes sense to make them mandatory